### PR TITLE
Detect if sed supports --in-place

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

Not all implementations of sed have the --in-place flag. The build system fails on these systems.

#9206 accidentally added one of the changes to the top level Makefile.am that should have been in this PR. This brings in the rest of the changes, fixing the dist-hook target.

### Description

Detect support for the flag during ./configure and provide a fallback mechanism for those systems where sed's behavior differs. The autoconf variable ${ac_inplace} can be used to choose the correct flags for editing a file in place with sed.

Replace violating usages in Makefile.am with ${ac_inplace}.

### How Has This Been Tested?

We have tested builds with this change on several distributions of Linux and on FreeBSD.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
I had previously closed a PR for this intending to detect and use GNU sed instead, but I'm reopening it to solve the immediate problems. Making the build depend on GNU sed can be done on top of these changes at a later time.

File: cmd/arcstat/Makefile.am [hunk 1]
```diff
@@ -8,6 +8,6 @@ dist_bin_SCRIPTS = arcstat
 #
 if USING_PYTHON_2
 install-exec-hook:
-	sed --in-place 's|^#!/usr/bin/env python3|#!/usr/bin/env python2|' \
+	sed ${ac_inplace} -e 's|^#!/usr/bin/env python3|#!/usr/bin/env python2|' \
 	    $(DESTDIR)$(bindir)/arcstat
```
Comment (chrisrd) [inline cmd/arcstat/Makefile.am]: What is the fallback mechanism? This looks like, if `${ac_inplace}` is empty, it's just going to print the modified `arcstat` to stdout.

Comment (freqlabs) [reply cmd/arcstat/Makefile.am]: `-i ''`

Comment (chrisrd) [reply cmd/arcstat/Makefile.am]: Ah, yes - sorry, I mis-read the patch as `--in-place` vs nothing, rather than `--in-place` vs `-i`.

**Pull request merged**